nf_core/lint.py

```diff
@@ -286,6 +291,43 @@ def check_readme(self):
                 self.failed.append((6, "Found a bioconda environment.yml file but no badge in the README"))
 
 
+    def check_version_consistency(self):
+        """ Check that the version variable value is numeric """
+        logging.debug('Checking if version is numeric')
+        versions = {}
+        # Get the version definitions
+        # First, from the TRAVIS_TAG env var
+        versions['travis_tag'] = os.environ.get('TRAVIS_TAG')
+        if not versions['travis_tag']:
+            self.failed.append((7, "Could not determine the release tag, TRAVIS_TAG env var was not set."))
+            return
+        
+        # Second, get version from nextflow.config
+        if not 'params.version' in self.config.keys():
+            self.failed.append((7, "Could not determine a pipeline version in the nextflow.config."))
+            return
+        versions['pipeline_version'] = self.config['params.version']
+
+        # Third, from the docker slug
+        if not 'process.container':
+            self.failed.append((7, "Could not determine a container version in the workflow configuration."))
+            return
+        versions['container_version'] = self.config['process.container'].split(':')[-1]
+
+        # Check if they are all numeric
+        for v_type, version in versions.items():
+            if not version.replace('.', '').isdigit():
+                self.failed.append((7, "{} was not numeric: {}!".format(v_type, version)))
+                return
+
+        # Check if they are consistent
+        if len(set(versions.values())) != 1:
+            self.failed.append((7, "The versioning is not consistent between container, release tag and config."))
```

Comment:
Would be nice to print the types and values in the error message..

Reply:
I agree